src/components/SignUp/Group/index.tsx

```diff
@@ -0,0 +1,244 @@
+  const [formData, setForm] = useForm(defaultData); // useForm is an extension of React hooks to manage form state
+
+  // default state for form validation error states per step
+  const [stepErrors, setStepErrors] = useState(
+    createDefaultStepErrorsObj(flatSteps(defaultSteps))
+  );
+
+  // console.log('formData', { formData });
+  // defaults for our defaultSteps
+  const { step, index, navigation } = useStep({
+    initialStep: currentStep,
+    steps: defaultSteps
+  });
+  const stepId = (step as Step).id;
+  // we are just manually assuming they've agreed to this right now
+  // remove once we clean this up
+  const setPrivacyAgree = () => {
+    const target = {
+      name: 'privacyAgreement',
+      value: true
+    };
+    setForm({ target });
+  };
+
+  // submit basics to Firebase, get response, set session
+  const submitBasics = async () => {
+    // we only get here if they've agreed to the privacy agreement
+    setPrivacyAgree();
+
+    const {
+      basicsFirstName,
+      basicsLastName,
+      basicsEmailAddress,
+      basicsPassword,
+      basics18Plus
+    } = formData;
+
+    await createUserWithEmailAndPassword(
+      firebaseAuth,
+      basicsEmailAddress,
+      basicsPassword
+    )
+      .then(async res => {
+        try {
+          const userId = res.user.uid;
+
+          const accountRef = await addDoc(
+            collection(firebaseFirestore, 'accounts'),
+            {
+              basics_18_plus: basics18Plus,
+              first_name: basicsFirstName,
+              last_name: basicsLastName,
+              privacy_agreement: true,
+              uid: userId
+            }
+          );
+
+          const profileRef = await addDoc(
+            collection(firebaseFirestore, 'profiles'),
+            {
+              uuid: userId,
+              account_id: accountRef.id
+            }
+          );
+
+          setAccountRef(accountRef);
+          setProfileRef(profileRef);
+        } catch (e) {
+          console.error('Error adding document:', e);
+        }
+      })
+      .catch(err => {
+        console.log('Error creating user:', err);
+      });
+  };
+
+  // callback function for updating if a step has errors
+  // we pass this down in the "hasErrorCallback" prop for the step
+  const setStepErrorsCallback = (step: string, hasErrors: boolean) => {
+    // console.log('step error', step, stepErrors);
+    const newStepErrorsObj = { ...stepErrors };
+
+    if (step in newStepErrorsObj) {
+      newStepErrorsObj[step] = hasErrors;
+    }
+
+    setStepErrors(newStepErrorsObj);
+  };
+
+  // based on which step we're on, return a different step component and pass it the props it needs
+  const stepFrame = () => {
+    // console.log('stepFrame', stepId);
+    const props = { formData, setForm, navigation };
+    let returnStep;
+
+    switch (stepId) {
+      case 'basics':
+        returnStep = (
+          <IndividualBasics
```

Comment:
I'm confused - this looks like all of the same stuff as the Individual sign-up? Is this just temporary to get the new folder structure in first and then you'll go back and add the group sign-up steps?

Reply:
Oh, duh, that's what this PR is haha